content_script.js

```diff
@@ -1,82 +1,213 @@
-var filteredTags = ["script", "noscript", "style"];//, "link"];
+/**
+ * HTML serializer that takes a document and synchronously stores it as an array of strings and stores enough state to later
+ * asynchronously convert it into an html text file.
+ */
+class HTMLSerializer { // TODO: BECAUSE MULTIPLE CONTENT SCRIPTS CAN BE INJECTED, GIVES Identifier 'HTMLSerializer' has already been declared ERROR
```

Comment:
Dont' use all-caps in TODOs. Also, Google style is to write TODO(sfine): do this and this.